### PR TITLE
New function development -- yurtctl adds parameter enable app manager to control automatic deployment of yurtappmanager.

### LINKED ISSUE #340: Yurtctl adds the parameter enable-app-manager to specify the deployment of yurt app manager.
**What would you like to be added**:
Yurtctl adds the parameter enable-app-manager.

**Why is this needed**:
To specify the deployment of yurt app manager.

**others**
/kind feature

### PR DESCRIPTION
**Pull request by yanyhui**
/kind feature

#### What this PR does / why we need it:
New function development.
#### Which issue(s) this PR fixes:

Fixes #340 

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

**Comment by yanyhui** [comment]
This function has been verified successfully.

**Comment by rambohe-ch** [comment]
@yanyhui Please fix golangci-lint error

File: pkg/yurtctl/util/kubernetes/util.go [hunk 1]
```diff
@@ -210,6 +218,145 @@ func CreateServiceFromYaml(cliSet *kubernetes.Clientset, svcTmpl string) error {
...
+			return err
+		}
+		unstructuredObj := &unstructured.Unstructured{Object: unstructuredMap}
+		gr, err := restmapper.GetAPIGroupResources(clientset.Discovery())
+		if err != nil {
+			return err
+		}
+
+		mapper := restmapper.NewDiscoveryRESTMapper(gr)
+		mapping, err := mapper.RESTMapping(gvk.GroupKind(), gvk.Version)
```
Comment (Peeknut) [inline pkg/yurtctl/util/kubernetes/util.go:335]: When was the gvk information of crd registered?

File: pkg/yurtctl/cmd/convert/convert.go [hunk 2]
```diff
@@ -377,6 +416,84 @@ func (co *ConvertOptions) RunConvert() (err error) {
...
 
+func deployYurtAppManager(
+	client *kubernetes.Clientset,
+	cloudNodes []string,
+	yurtappmanagerImage string,
+	yurtAppManagerClient dynamic.Interface) error {
+
+	// 1.create the YurtAppManagerCustomResourceDefinition
+	// 1.1 nodepool
+	if err := kubeutil.CreateCRDFromYaml(client, yurtAppManagerClient, "",[]byte(constants.YurtAppManagerNodePool)); err != nil {
```
Comment (Peeknut) [inline pkg/yurtctl/cmd/convert/convert.go:427]: It seems like the func `CreateCRDFromYaml` is to create a cr. However, it's used to register crd here.

File: pkg/yurtctl/util/kubernetes/util.go [hunk 3]
```diff
@@ -357,6 +504,23 @@ func GenClientSet(flags *pflag.FlagSet) (*kubernetes.Clientset, error) {
 	return kubernetes.NewForConfig(restCfg)
 }
 
+//add by yanyhui at 20210611
```
Comment (Peeknut) [inline pkg/yurtctl/util/kubernetes/util.go:507]: It is better to delete the comment.

File: pkg/yurtctl/cmd/convert/convert.go [hunk 4]
```diff
@@ -132,6 +136,11 @@ func NewConvertCmd() *cobra.Command {
 	cmd.Flags().String("pod-manifest-path",
 		"/etc/kubernetes/manifests",
 		"Path to the directory on edge node containing static pod files.")
+	cmd.Flags().BoolP("enable-app-manager", "e", false,
+		"if set, yurtappmanager will be deployed.")
+	cmd.Flags().String("yurt-app-manager-image",
+		"openyurt/yurt-app-manager:v0.4.0",
```
[the remark this answers is not in the document]
Comment (yanyhui) [reply pkg/yurtctl/cmd/convert/convert.go:142]: The latest version image can't be used without updating. Now we all use the version of v0.4.0.

File: pkg/yurtctl/cmd/convert/convert.go [hunk 5]
```diff
@@ -353,6 +380,18 @@ func (co *ConvertOptions) RunConvert() (err error) {
 	if err != nil {
 		return err
 	}
+	
+	//8. deploy the yurtappmanager if required
```
[the remark this answers is not in the document]
Comment (yanyhui) [reply pkg/yurtctl/cmd/convert/convert.go:384]: Step 7 and "deploy the yurtappmanager" are separate processes and have no impact, so I don't think we need to put them before step 7.

File: pkg/yurtctl/cmd/convert/convert.go [hunk 6]
```diff
@@ -377,6 +416,84 @@ func (co *ConvertOptions) RunConvert() (err error) {
 	return
 }
 
+func deployYurtAppManager(
+	client *kubernetes.Clientset,
+	cloudNodes []string,
```
[the remark this answers is not in the document]
Comment (yanyhui) [reply pkg/yurtctl/cmd/convert/convert.go:421]: This is a good idea. I found that this parameter was not used at that time.

File: pkg/yurtctl/cmd/convert/convert.go [hunk 7]
```diff
@@ -38,6 +38,7 @@ import (
 	"github.com/openyurtio/openyurt/pkg/yurtctl/lock"
 	kubeutil "github.com/openyurtio/openyurt/pkg/yurtctl/util/kubernetes"
 	strutil "github.com/openyurtio/openyurt/pkg/yurtctl/util/strings"
+	"k8s.io/client-go/dynamic"
```
[the remark this answers is not in the document]
Comment (yanyhui) [reply pkg/yurtctl/cmd/convert/convert.go:41]: modified

File: pkg/yurtctl/util/kubernetes/util.go [hunk 8]
```diff
@@ -50,6 +50,14 @@ import (
 	"github.com/openyurtio/openyurt/pkg/yurtctl/constants"
 	strutil "github.com/openyurtio/openyurt/pkg/yurtctl/util/strings"
 	tmplutil "github.com/openyurtio/openyurt/pkg/yurtctl/util/templates"
+	"bytes"
```
[the remark this answers is not in the document]
Comment (yanyhui) [reply pkg/yurtctl/util/kubernetes/util.go:53]: modified

File: pkg/yurtctl/util/kubernetes/util.go (hunk 3 above)
Comment (yanyhui) [reply pkg/yurtctl/util/kubernetes/util.go:507]: delete

File: pkg/yurtctl/util/kubernetes/util.go [hunk 9]
```diff
@@ -210,6 +218,145 @@ func CreateServiceFromYaml(cliSet *kubernetes.Clientset, svcTmpl string) error {
...
+	}
+	klog.V(4).Infof("validatingwebhookconfiguration/%s is created", svc.Name)
+
+	return nil
+}
+
+func CreateCRDFromYaml(clientset *kubernetes.Clientset, yurtAppManagerClient dynamic.Interface, nameSpace string,filebytes []byte) error {
+	var err error
+	decoder := yamlutil.NewYAMLOrJSONDecoder(bytes.NewReader(filebytes), 10000)
+	for {
```
[the remark this answers is not in the document]
Comment (yanyhui) [reply pkg/yurtctl/util/kubernetes/util.go:318]: delete

File: pkg/yurtctl/util/kubernetes/util.go [hunk 10]
```diff
@@ -210,6 +218,145 @@ func CreateServiceFromYaml(cliSet *kubernetes.Clientset, svcTmpl string) error {
...
+		var rawObj runtime.RawExtension
+		if err = decoder.Decode(&rawObj); err != nil {
+			break
```
[the remark this answers is not in the document]
Comment (yanyhui) [reply pkg/yurtctl/util/kubernetes/util.go:321]: modified

File: pkg/yurtctl/util/kubernetes/util.go (hunk 1 above)
Comment (yanyhui) [reply pkg/yurtctl/util/kubernetes/util.go:335]: Before creating CRD.

File: pkg/yurtctl/cmd/convert/convert.go (hunk 2 above)
Comment (yanyhui) [reply pkg/yurtctl/cmd/convert/convert.go:427]: register crd

**Comment by kadisi** [comment]
/approve 

thanks @yanyhui

**Pull request merged**